packages/controller/src/resources/tenants/index.ts

```diff
@@ -15,10 +15,53 @@
  */
 
 import { KubeConfig } from "@kubernetes/client-node";
-import { ResourceCollection, Namespace } from "@opstrace/kubernetes";
+import { ResourceCollection, Namespace, Secret } from "@opstrace/kubernetes";
 import { State } from "../../reducer";
 import { getTenantNamespace } from "../../helpers";
 
+const CERT_SECRET_NAMESPACE = "ingress"
+const CERT_SECRET_NAME = "https-cert"
+
+const getCertSecretCopy = (targetNameSpace: string,  state: State, kubeConfig: KubeConfig) => {
+  const certSecretSource = state.kubernetes.cluster.Secrets.resources.find(
+    secret =>
+      secret.name === CERT_SECRET_NAME &&
+      secret.namespace === CERT_SECRET_NAMESPACE
+  );
+
+  if (!certSecretSource) {
+    // TODO: Is this a situation that needs to be handled, and if so, how?
```

Comment:
Yes, when the cluster is created and the controller is installed this will always throw an error. 

The controller [creates a list of all the resources it needs to create](https://github.com/opstrace/opstrace/blob/main/packages/controller/src/tasks/reconciliation.ts#L75-L89) and sends the create requests in batches. The first time it runs this function it'll always error because the source secret is not created yet.

Reply:
So I just added a [log output](https://github.com/opstrace/opstrace/pull/749/files#diff-8f6735699198039ebd9c077e47a7145816d8760f9f09eb635896a5ba2ec9effaR37) for right now. Let me know if the log copy can be improved.